rules-reviewed/eap7/weblogic/weblogic-webapp.windup.xml

```diff
@@ -125,7 +141,13 @@
                 <hint title="WebLogic ApplicationLifecycleEvent" effort="3" category-id="mandatory">
                     <message>
                       <![CDATA[
-                      WebLogic `ApplicationLifecycleEvent` must be replaced with standard Java EE `ServletContextEvent`. Otherwise, a custom solution using CDI's `ApplicationScoped` beans or EJB's `@Startup` beans is required in order to propagate a custom event object because `ServletContextEvent` types are not extendible in the standard Java EE programming model.
+                      WebLogic `ApplicationLifecycleEvent`s are not suitable for migration. 
+                      
+                      Recommendation:
+                      
+                      Replace each proprietary WebLogic `ApplicationLifecycleEvent` with a standard Java EE  `ApplicationLifecycleEvent.
+
+                      A less preferred option is to implement a custom solution using CDI's `ApplicationScoped` beans or EJB's `@Startup` beans in order to propagate a custom event object because `ServletContextEvent` types are not extendible in the standard Java EE programming model.
```

Comment:
I am reading "otherwise" in the original context as meaning "a less good option" . Who can confirm? I know we generally avoid options anyway -- does that principle apply here?